src/Config.php

```diff
@@ -66,6 +67,8 @@ public function __construct(array $config = [])
                 unset($config[$key]);
             }
         }
+
+        /** @psalm-suppress ArgumentTypeCoercion */
```

Comment:
This coercion shouldn't apply anymore: `merge()` has the correct types? 🤔

Reply:
`ArrayUtils::merge` returns `array` and thus, psalm loses track.

Not sure if something like this would help psalm to keep track:

```
/**
 * @template TArray1 of array
 * @template TArray2 of array
 * @param TArray1 $a
 * @param TArray2 $b
 * @return TArray1&TArray2
 */
```

But this wont properly work for nested arrays (as they are here). There is absolutely no way but to trust that `ArrayUtils::merge` will return the same types.

Reply:
Yes, but you are not calling `ArrayUtils::merge()`, but rather `$this->merge()`, which is well typed, and has its own suppressions

Reply:
I'll leave this alone for now - I think it's not worth wasting effort on it after all the work you've already put in this, and it's in the internals anyway :)

Reply:
I've added `reportUnusedPsalmSuppress` in the config and thats not reported so something might still be odd :-/